site-builder/src/site/resource.rs

```diff
@@ -399,6 +384,43 @@ impl ResourceManager {
         )))
     }
 
+    /// Based on the ws-resources.yaml definition, this function derives the HTTP headers
+    /// based on wildcard matching, the headers that should be added to the HTTP response
+    /// of the resource defined by the resource_path.
+    fn derive_http_headers(&self, resource_path: &str) -> BTreeMap<String, String> {
+        self.ws_resources
+            .as_ref()
+            .and_then(|config| config.headers.as_ref())
+            .and_then(|headers| {
+                headers
+                    .iter()
+                    .filter(|(path, _)| {
+                        // Replace the wildcard with a regex wildcard.
+                        let path_regex = path.replace('*', ".*");
+                        // Check if the resource_path matches the path_regex.
+                        match Regex::new(&path_regex) {
+                            Ok(re) => re.is_match(resource_path),
+                            Err(_) => false,
+                        }
+                    })
+                    .max_by_key(|(path, _)| path.len())
+                    .map(|(_, header_map)| header_map)
+                    .into_iter()
+                    .reduce(|_, header_map| header_map)
```

Comment:
Question: why do you need the reduce here?

Reply:
After revisiting this it seems that is not needed! 😅 
Updated it, thanks for pointing it out